packages/relay/src/lib/eth.ts

```diff
@@ -568,34 +674,43 @@ export class EthImpl implements Eth {
    * @param hash
    */
   async getTransactionReceipt(hash: string) {
-    // FIXME: This should go to the mirror node. For now we have an in memory map we
-    //        grab receipts from for each tx that was executed. Even in the full implementation
-    //        we will want to use the cache as an LRU to avoid hammering the mirror node more
-    //        than needed, since the receipt is an immutable type. But the cache should be in the
-    //        mirror node connector, rather than here.
-    this.logger.trace('getTransactionReceipt(hash=%s)', hash);
-    try {
-      // Lookup the receipt in the cache.
-      const receipt = cache.get(hash);
-
-      if (!receipt) {
-        // FIXME: if it wasn't in the cache, go to the mirror node. If it isn't there either,
-        //        then return null.
-        this.logger.debug(
-          'Did not find the receipt for hash %s. Returning null.',
-          hash
-        );
-        return null;
-      }
-
-      this.logger.trace('Found a receipt, returning it to the caller');
-      return receipt;
-    } catch (e) {
-      this.logger.error(
-        e,
-        'Failed to handle getTransactionReceipt cleanly for hash %s',
-        hash
-      );
+    this.logger.trace(`getTransactionReceipt(${hash})`);
+    const receiptResponse = await this.mirrorNodeClient.getContractResult(hash);
+    this.logger.trace(`response - ${JSON.stringify(receiptResponse)}`);
+    if (receiptResponse === undefined) {
```

Comment:
This will never be true.
At worst it's null or receiptResponse.hash is undefined.

```suggestion
    if (receiptResponse === null || receiptResponse.hash === undefined) {
```

Likely causing your test failure

Reply:
done